tests/algorithms/test_adam.py

```diff
@@ -0,0 +1,156 @@
+"""Tests for Adam."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+import torch.nn.functional as F
+
+from mrpro.algorithms import adam
+from mrpro.data import SpatialDimension
+from mrpro.operators import ConstraintsOp
+from mrpro.operators.functionals import L2_data_discrepancy
+from tests import RandomGenerator
+from tests.operators._OptimizationTestFunctions import Rosenbrock
+from tests.phantoms._SaturationRecoveryEllipsePhantom import SaturationRecoveryEllipsePhantom2D
+
+
+@pytest.mark.parametrize(
+    'a, b',
+    [(1.0, 100.0), (2.0, 50.0), (5.0, 10.0), (10.0, 5.0)],
+)
+def test_adam_rosenbrock(a, b):
```

Comment:
this is pretty much the same for adam and lbfgs, right?
can we make it one test parameterized by the optimisation function or at least refactor it such that we don't have that much duplicate code?